x-pack/plugin/data-frame/src/test/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexerTests.java

```diff
@@ -232,8 +232,9 @@ public void testPageSizeAdapt() throws InterruptedException {
             assertThat(indexer.getState(), equalTo(IndexerState.STARTED));
             assertTrue(indexer.maybeTriggerAsyncJob(System.currentTimeMillis()));
             assertThat(indexer.getState(), equalTo(IndexerState.INDEXING));
+
             latch.countDown();
-            awaitBusy(() -> indexer.getState() == IndexerState.STOPPED);
+            assertBusy(() -> assertThat(indexer.getState(), equalTo(IndexerState.STOPPED)));
```

Comment:
Pinging @elastic/ml-core - I've changed this line from `awaitBusy()` to `assertBusy()`, but since the latter requires the assertion to pass, the test now fails. I tried upping the timeout to 30s but that didn't help. Can you advise?

Reply:
@benwtrent I think you are right, looks like a bug in the test.

As for explaining the test: The test uses a mock indexer which throws a circuit breaker exception on every search, the failure is handled and the indexer should change its state from `INDEXING` to `STARTED` (not `STOPPED`).

Funny this worked before. @pugnascotia can you try the suggestion? Same applies to line 251 below.

Reply:
The suggestion change passes locally, I've pushed a fix. Thanks for the help 👍